src/Gateway/ScheduledDataflowGateway.php

```diff
@@ -6,54 +6,50 @@
 
 use CodeRhapsodie\DataflowBundle\Entity\ScheduledDataflow;
 use CodeRhapsodie\DataflowBundle\Repository\ScheduledDataflowRepository;
-use Doctrine\ORM\EntityManagerInterface;
 
 class ScheduledDataflowGateway
 {
-    /** @var EntityManagerInterface */
-    private $em;
     /** @var ScheduledDataflowRepository */
     private $scheduledDataflowRepository;
 
-    public function __construct(EntityManagerInterface $em, ScheduledDataflowRepository $scheduledDataflowRepository)
+    public function __construct(ScheduledDataflowRepository $scheduledDataflowRepository)
     {
-        $this->em = $em;
         $this->scheduledDataflowRepository = $scheduledDataflowRepository;
     }
 
-    public function find(int $id): ScheduledDataflow
+    public function find(int $id): ?ScheduledDataflow
     {
         return $this->scheduledDataflowRepository->find($id);
     }
 
     public function findAllOrderedByLabel(): iterable
     {
-        return $this->scheduledDataflowRepository->findBy([], ['label' => 'asc']);
+        $qb = $this->scheduledDataflowRepository->createQueryBuilder();
+        $qb->orderBy('label', 'asc');
+
+        $stmt = $qb->execute();
+        if (0 === $stmt->rowCount()) {
+            return [];
+        }
+
+        return $stmt->fetchAll(\PDO::FETCH_ASSOC);
```

Comment:
The return value was an iterable of `ScheduledDataflow` before, we should keep the same return type.

Add the docblock with `@return iterable|ScheduledDataflow[]` to make this clearer.

Reply:
The same reason for JobGateway.

Reply:
What is the goal?

Reply:
This function is not optimal because the equivalent in `JobGateway` is `getOneshotListQueryForAdmin` and return a `QueryBuilder` to use in the pager.

For `ScheduledDataflowGateway::findAllOrderedByLabel` no pager is used, why?

Reply:
Pager has been added